### ISSUE TITLE
why is rustlings lsp not on by default?

### ISSUE THREAD

**Issue by xdavidliu** [rust-lang/rustlings#1351]
I was doing exercises using VSCode and rust-analyzer, and was confused and frustrated for nearly 30 minutes about why rust-analyzer was not working for my exercises. I thought my VScode and rust-analyzer were broken, and went down many false paths, especially since there are already many other cases of rustlings-analyzer being completely broken (which was *not* the case here!)

Only after more than 30 min of googling did I realize that I needed to opt IN to this by doing `rustlings lsp`. What a horrible user-unfriendly experience! How are new users supposed to know that they need to do this command in order to get this to work?

Any way to just remove the step of requiring to `rust-analyzer lsp`, so that it simply works out of the box? This extra step is very, very easy to miss for rustlings' target audience, and really left a bad impression when I ran into it.

edit: if it's not practical to remove the requirement to do `rust-analyzer lsp`, one alternative I would be okay with is to more loudly advertise this fact; this would save a lot of confusion and frustration for rustlings target audience.

**Comment by diannasoreil** [comment]
AFAIK not, as we don't follow the typical Cargo project structure, which is why we adopted `rustlings lsp` in the first place. The `rust-project.json` file needs a path to your Rust installation, which is why we can't check it into the repository (someone else with rust-analyzer internal knowledge, feel free to prove me wrong).

How to enable rust-analyzer is explained [here](https://github.com/rust-lang/rustlings#enabling-rust-analyzer). This is already in the README file, which should be your first go-to destination if you need documentation about the project. I don't think I want to advertise it in any more obvious way, and it doesn't seem to be that big of an issue.

**Comment by jamt9000** [comment]
I had a similar experience as a first-time rust user being extremely confused why features of the vscode extension like inlay hints seemed to just not work (and lots of google dead ends until I found this issue - so thank you @xdavidliu!). Maybe this is a shortcoming of the rust ecosystem in general, but since it's a teaching tool for beginners I would have appreciated a very visible notice about the need to run `rustlings lsp`.

After installing I continued by following the friendly instructions from the rustlings command itself rather than reading all the github README (and I assumed it was a vscode issue so didn't think to read the rustlings docs). So perhaps it would be good to show a notice when running `rustlings` or `rustlings watch` about the need to run `rustlings lsp`.
